### PR TITLE
Add a worker subpackage

### PR DESCRIPTION
**Pull request by collijk**
### Description

- *Category*: refactor
- *JIRA issue*: [MIC-2881](https://jira.ihme.washington.edu/browse/MIC-2881)

Split the worker side RQ database logic from the worker simulation execution
logic in a worker subpackage and extract the code that produces the worker 
launch script from the drmaa job template initialization. This also sets up the 
structure to parameterize  psimulate with test workloads (see [MIC-2357](https://jira.ihme.washington.edu/browse/MIC-2357)),
which is how I'll do a first round integration test of all this shuffling around.

### Testing
CI

### CODE REVIEW

**Review by albrja — APPROVED** [review]

File: src/vivarium_cluster_tools/psimulate/runner.py [hunk 1]
```diff
@@ -187,6 +188,13 @@ def main(
         num_jobs=len(job_parameters),
         redis_logging_root=output_paths.logging_root,
     )
+    # Spin up a unified interface to all the redis databases
+    registry_manager = redis_dbs.RegistryManager(redis_ports, num_jobs_completed)
+    # Distribute all the remaining jobs across the job queues
+    # in the redis databases.
+    registry_manager.enqueue(
+        jobs=job_parameters, workhorse_import_path=worker.WORK_HORSE_PATHS["vivarium"]
```
Comment (rmudambi) [inline src/vivarium_cluster_tools/psimulate/runner.py:196]: Nit: can "vivarium" ultimately become a constant?

File: src/vivarium_cluster_tools/psimulate/runner.py [hunk 2]
```diff
@@ -24,6 +24,7 @@
     programming_environment,
     redis_dbs,
     results,
+    worker,
```
Comment (rmudambi) [inline src/vivarium_cluster_tools/psimulate/runner.py:27]: Is it possible to get typing here? I think it'd make the code more readable.

File: src/vivarium_cluster_tools/psimulate/redis_dbs/registry.py [hunk 3]
```diff
@@ -68,13 +66,13 @@ def completed(self):
     def jobs_to_finish(self):
         return not (self.failed or self.completed)
 
-    def enqueue(self, jobs):
+    def enqueue(self, jobs, workhorse_import_path: str):
```
Comment (rmudambi) [inline src/vivarium_cluster_tools/psimulate/redis_dbs/registry.py:69]: Can we specify the type of jobs?

File: src/vivarium_cluster_tools/psimulate/worker/core.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+                super(_ResilientWorker, self).work(*args, **kwargs)
+                return
+            except redis.exceptions.ConnectionError:
+                backoff = random.random() * 60
+                logger.error(f"Couldn't connect to redis. Retrying in {backoff}...")
+                retries += 1
+                time.sleep(backoff)
+        logger.error(f"Ran out of retries. Killing worker")
+
+    def main_work_horse(self, job, queue):
```
Comment (rmudambi) [inline src/vivarium_cluster_tools/psimulate/worker/core.py:104]: Typing?

**Review by rmudambi — APPROVED** [review]

File: src/vivarium_cluster_tools/psimulate/runner.py (hunk 1 above)
Comment (collijk) [reply src/vivarium_cluster_tools/psimulate/runner.py:196]: For sure.  It's in a lot of places and I don't remember the context for all of them.  I plan to address in the load_test jobs bucket of work.

**Comment by collijk** [comment]
I'll make a typing PR across the whole of the repo shortly.

**Pull request merged**